Devscord.DiscordFramework/Framework/Commands/Services/BotCommandsMatchingService.cs

```diff
@@ -1,87 +1,154 @@
+        private bool AreAllGivenArgsForCommandKnown(IEnumerable<BotCommandProperty> properties, IEnumerable<DiscordRequestArgument> arguments)
         {
-            var notOptionalCount = properties.Count(x => !x.IsOptional);
-            var parametersCount = arguments.Count(x => !string.IsNullOrEmpty(x.Name));
-            if (parametersCount > properties.Count || parametersCount < notOptionalCount)
+            var argumentsNames = arguments
+                .Select(x => x.Name?.ToLowerInvariant())
+                .ToList();
+            var lists = properties
+                .Where(property => property.Type == BotCommandPropertyType.List)
+                .Select(x => x.Name.ToLowerInvariant())
+                .ToList();
+            var argsWithoutListSubarguments = this.GetArgsWithoutListSubarguments(argumentsNames, lists);
+            return argsWithoutListSubarguments.All(arg => properties.Any(property => arg == property.Name.ToLowerInvariant()));
+        }
+
+        private List<string> GetArgsWithoutListSubarguments(List<string> argumentsNames, List<string> listsNames)
+        {
+            for (int i = 0; i < argumentsNames.Count; i++)
             {
-                return false;
+                if (!listsNames.Contains(argumentsNames[i]))
+                {
+                    continue;
+                }
+                while (i + 1 < argumentsNames.Count && argumentsNames[i + 1] == null)
+                {
+                    argumentsNames.RemoveAt(i + 1);
+                }
             }
-            foreach (var argument in arguments)
+            return argumentsNames;
+        }
+
+        private void CheckQuotationMarksInListsAndTexts(IEnumerable<BotCommandProperty> properties, IEnumerable<KeyValuePair<string, string>> argsAndValues, bool isCommandMatchedWithCustom = false)
+        {
+            foreach (var property in properties)
             {
-                var matchedByName = properties.FirstOrDefault(property => argument.Name?.ToLowerInvariant() == property.Name.ToLowerInvariant());
-                if (matchedByName == null)
+                if (property.Type != BotCommandPropertyType.Text && property.Type != BotCommandPropertyType.List)
                 {
                     continue;
                 }
-                if (matchedByName.Type == BotCommandPropertyType.Bool)
+                var argumentAndValue = argsAndValues.FirstOrDefault(arg => arg.Key?.ToLowerInvariant() == property.Name.ToLowerInvariant());
+                if (string.IsNullOrWhiteSpace(argumentAndValue.Value))
                 {
                     continue;
                 }
-                if (matchedByName.Type == BotCommandPropertyType.List && !string.IsNullOrEmpty(argument.Value))
+                if (isCommandMatchedWithCustom && argumentAndValue.Value.Count(x => x == '\"') % 2 != 0)
```

Comment:
a dlaczego to jest sprawdzane tylko jeśli jest isCommandMatchedWithCustom?

Reply:
tutaj masz przypadek dla customowych komend, a niżej jeszcze jest przypadek dla default command. W tym przypadku wystarczy tylko sprawdzić parzystość cudzysłowów, aby stwierdzić czy jest odpowiednia ilość tych znaków. W przypadku domyślnych trzeba niestety dla każdego argumentu sprawdzać czy nie ma cudzysłowia (ogólnie CommandParser usuwa te znaki chyba, że jest ich nieparzysta ilość wtedy coś zostaje).